### ISSUE TITLE
[DATA] Baselines definitions

### ISSUE THREAD

**Issue by ZDev-19** [unmsmfisi-socialapplication/social_app#592]
### **Objective**

Document all baselines politics and agreements in DATA features with QA Team

**Comment by ZDev-19** [comment]
### **Alcance**

- Definicion de las politicas para el testeo en modelos de Data por parte de QA
- Acordar las baselines correspondientes para la auditoria de modelos
- Discutir las alternativas para la seleccion de baselines

**Comment by ZDev-19** [comment]
### **Testing for Data models*

**Comment by ZDev-19** [comment]
### **Model Baselines**

**Comment by ZDev-19** [comment]
### **QA agreements**

**Comment by ZDev-19** [comment]
### **Baseline Model and evaluation development**

With the approval of QA members , the following development is implemented in Data folder features:

- Baseline model development (DummyClassifier)
- Baseline rates metrics implementation with training data (Zero and random rate)

> ### **Dummy Classifier Model**

The library Scikit-L;earn givers a perfect option for development of a baseline model , named "DummyClasifier" it creates a low complex model with simple desicion making , it also has different baseline approaches:

<html>

  <table>
    <tr>
      <th>Approach</th>
      <th>Description</th>
    </tr>
    <tr>
      <td>stratified</td>
      <td>This strategy randomly selects class labels based on the class distribution in the training set. It aims to maintain the same class distribution as the training data, making it useful for imbalanced classes.</td>
    </tr>
    <tr>
      <td>most_frequent</td>
      <td>This strategy always predicts the most frequent class in the training set. It is suitable for imbalanced datasets where the majority class dominates the distribution. It provides a baseline performance level based on the most common class, without considering input features.</td>
    </tr>
    <tr>
      <td>uniform</td>
      <td>This strategy assigns class labels randomly and uniformly, without considering the class distribution in the training data. It is useful when there is no specific pattern or information to guide the predictions.</td>
    </tr>
    <tr>
      <td>constant</td>
      <td>This strategy always predicts a constant class label specified by a constant parameter. It helps create a baseline model that consistently predicts a particular class. It aids in evaluating the impact of class imbalance and comparing model performance against a fixed prediction.</td>
    </tr>
  </table>

</body>

</html>

**Comment by ZDev-19** [comment]
### **Implementation Issue**

After the discussion with QA members @luzelenatt @felixalfonsofernandez and @PaulaRV01 , the baselines methods were defined properly . This issue represents all the topics discussed in the meeting

> **Development**

The issue for integrating all baseline testing and metrics is , it will contemplate the results of the implementation and testing in each key feature model

**Comment by ZDev-19** [comment]
### **References**

- https://medium.com/@preethiprakash29/understanding-baseline-models-in-machine-learning-3ed94f03d645
- https://towardsdatascience.com/calculating-a-baseline-accuracy-for-a-classification-model-a4b342ceb88f
- https://censius.ai/blogs/how-to-implement-baselines-in-ml-modeling
- https://www.youtube.com/watch?v=wfTk7Lb9uPg
